app/main/authorize/access_token_sign_in_required.py

```diff
@@ -40,10 +41,25 @@ def decorated_view(*args, **kwargs):
         g.access_token_sign_in_required = True  # Set attribute on g
 
         try:
-            if not _validate_access_token():
+            access_token = session["access_token"]
+            refresh_token = session["refresh_token"]
+            try:
+                (
+                    access_token,
+                    refresh_token,
+                    tokens_are_refreshed,
+                ) = _validate_or_refresh_tokens(access_token, refresh_token)
```

Comment:
_validate_or_refresh_tokens doesnt concern itself with sessions.

Did consider just testing all the edge cases in that function itself, making it a public function but after tidying up the test cases I think its just about okay how it is, if it got any bigger I would do that.